src/Build.UnitTests/BinaryLogger_Tests.cs

```diff
@@ -24,67 +24,54 @@ public class BinaryLoggerTests
                <Exec Command='echo a'/>
             </Target>
          </Project>";
+        private readonly TestEnvironment _env;
+        private string _logFile;
 
-        [Fact]
-        public void TestBinaryLoggerRoundtrip()
+        public BinaryLoggerTests(ITestOutputHelper output)
         {
-            var originalTargetOutputLogging = Environment.GetEnvironmentVariable(MSBUILDTARGETOUTPUTLOGGING);
+            _env = TestEnvironment.Create(output);
 
-            try
-            {
-                var binaryLogger = new BinaryLogger();
+            // this is needed to ensure the binary logger does not pollute the environment
+            _env.WithEnvironmentInvariant();
+
+            _logFile = _env.ExpectFile(".binlog").Path;
+        }
 
-                // with this file name, the file will be archived as build artifact so we can inspect it later
-                // this is needed to investigate an intermittent failure of this test on Ubuntu 14
-                var logFilePath = "Microsoft.Build.Engine.UnitTests.dll_TestBinaryLoggerRoundtrip.binlog";
-                binaryLogger.Parameters = logFilePath;
+        [Fact]
+        public void TestBinaryLoggerRoundtrip()
+        {
+            var binaryLogger = new BinaryLogger();
 
-                var mockLogger1 = new MockLogger();
+            binaryLogger.Parameters = _logFile;
 
-                // build and log into binary logger and mockLogger1
-                ObjectModelHelpers.BuildProjectExpectSuccess(s_testProject, binaryLogger, mockLogger1);
+            var mockLogger1 = new MockLogger();
 
-                var mockLogger2 = new MockLogger();
+            // build and log into binary logger and mockLogger1
+            ObjectModelHelpers.BuildProjectExpectSuccess(s_testProject, binaryLogger, mockLogger1);
 
-                var binaryLogReader = new BinaryLogReplayEventSource();
-                mockLogger2.Initialize(binaryLogReader);
+            var mockLogger2 = new MockLogger();
 
-                // read the binary log and replay into mockLogger2
-                binaryLogReader.Replay(logFilePath);
+            var binaryLogReader = new BinaryLogReplayEventSource();
+            mockLogger2.Initialize(binaryLogReader);
 
-                if (File.Exists(logFilePath))
-                {
-                    File.Delete(logFilePath);
-                }
+            // read the binary log and replay into mockLogger2testassembly
+            binaryLogReader.Replay(_logFile);
 
-                Assert.Equal(mockLogger1.FullLog, mockLogger2.FullLog);
-            }
-            finally
-            {
-                Environment.SetEnvironmentVariable(MSBUILDTARGETOUTPUTLOGGING, originalTargetOutputLogging);
-            }
+            Assert.Equal(mockLogger1.FullLog, mockLogger2.FullLog);
         }
 
         [Fact]
-        public void NonExistingDirectory()
```

Comment:
Are you removing the NonExistingDirectory() test?

Reply:
Its asserts are being replaced by the new TestEnvironments checks, which makes the test obsolete.